Gems/Atom/Feature/Common/Assets/Shaders/PostProcessing/BloomCompositeCS.azsl

```diff
@@ -102,7 +102,11 @@ void MainCS(uint3 dID : SV_DispatchThreadID)
         PassSrg::m_inputTexture.SampleLevel(PassSrg::LinearSampler, sourceUv, PassSrg::m_sourceMipLevel).rgb;    
     float3 color1 = PassSrg::m_outputTexture[dID.xy].xyz;
 
-    float3 color = color0 * PassSrg::m_intensity * PassSrg::m_tint + color1;
+    float3 color = color1;
+    if (color0.x == color0.x && color0.y == color0.y && color0.z == color0.z )
+    {
+        color += color0 * PassSrg::m_intensity * PassSrg::m_tint;
+    }
```

Comment:
The issue "seems" like it is fixed because it is skipping this if statement as comparing floats is never a good idea. If you really need to compare floats you check if the difference between them is lower than some small epsilon but that is not advisable here. The issue is probably with color0. Try to go up the stack and start eliminating code that calculates it and figure out where the NAN is coming from.

Reply:
Color0 has Nans here, and comparing is to check for not-a-number.  However, I do not have the expertise to understand go through the previous steps, but I hope it can help find real solution.